BEP7.md

```diff
@@ -0,0 +1,320 @@
+  - [6. Token Specification](#6-token-specification)
+    - [6.1 Token Properties](#61-token-properties)
+    - [6.2 Token Management Operation](#62-token-management-operation)
+      - [6.2.1 Issue token](#621-issue-token)
+      - [6.2.2 Transfer Tokens](#622-transfer-tokens)
+      - [6.2.3 Freeze Tokens](#623-freeze-tokens)
+      - [6.2.4 Unfreeze Tokens](#624-unfreeze-tokens)
+  - [7. Token Use Case Examples](#7-token-use-case-examples)
+  - [8. License](#8-license)
+  - [9. Reference](#9-reference)
+
+## 1. Summary
+
+This BEP describes a proposal for non-fungible tokens(NFTs) management on the Binance Chain.
+
+## 2. Abstract
+
+BEP-7 Proposal describes a common set of rules for non-fungible token within the Binance Chain ecosystem. It introduces the following details of a non-fungible token on Binance Chain:
+
+- What information makes a non-fungible token on Binance Chain
+- What actions can be performed on a non-fungible token on Binance Chain
+
+### Couple things to keep in mind
+
+- This BEP won't contain "how to implement" in terms of implmentation on the Binance Chain codebase, e.g. which module NFT should be in (this kind of discussion is happening on [cosmos-sdk/issues/4046](https://github.com/cosmos/cosmos-sdk/issues/4046)), because that should be happening in the PR(against to Binance Chain, when it's fully open source, so that dev can send out PR.
+
+- Apparently Binance chain doens't have solidity-such smart contract implemented, so don't expect to see many smart contract implmenations like what you can see in [EIP-721](https://github.com/ethereum/EIPs/blob/master/EIPS/eip-721.md) in this BEP.
+
+## 3. Status
+
+This BEP is under drafting.
+
+## 4. Motivation
+
+Design and issue non-fungible asset on the Binance Chain, as one of the basic economic foundations of the blockchain.
+
+## 5. Collection Specification
+
+### 5.1 Collection Properties
+
+- Owner: the owner adress of the collection.
+
+- Token Name: Token Name represents the long name of the token - e.g. "BinanceAnnualEvent2019"
+
+- Denom: Denom is the identifier(symbol) of the newly issued collection.
+
+- Total Supply: Total supply is the total number of issued NFT tokens belong to this collection.
+
+- Mintable: A boolean value to indicate whether the owner can increase the totalSupply.
+
+- NFTs: An array contains items in NFT interface(described below).
+
+```go
+// Collection of non fungible tokens
+type Collection struct {
+  Owner         string  `json:"owner"`                    // owner of the collection;
+  Name          string  `json:"name"`                     // name of the collection;
+  Denom         string  `json:"denom,string,omitempty"`   // the identifier(symbol) of the newly issued collection;
+  TotalSupply   int64   `json:"totalSupply"`              // total supply of the NFT in this collection;
+  Mintable      bool    `json:"mintable"`                 // can increase total supply or not;
```

Comment:
Is this adequate? Doesn't it also need information about who can mint etc? Is it relevant only for BEP native NFTs?

Reply:
Yes, this is only relevant to BEP native NFT's @amazingandyyy I think he specified this within the BEP2 spec for native BNB tokens as they were ERC-20 but were made to be specific to the Binance Chain. 
I think that the approach used in Smart Contract ERC-721 and ERC-1155 is to use inheritance to delegate control of the Ownership Contract, so maybe perhaps the structure of the Ownership Contract would be in the Mint Type function and thus making Ownership a parameter that is always set to Owned by origin chain and as ownership is transferred between chain A and chain B then the host(origin) chain can delegate that Ownership responsibility to the receiving chain to then Transact or Trade on its behalf and have all the functionality of the Ownership chain except for the TotalSupply. This could be part of the on-chain metadata packet that gets sent along with Ownership permissions. In a BEP native chain, this would allow ultimate cross-chain reliability of that metadata to be read by any wallet that can read permissions. They could go freely between Smart Contract based wallets as they would have met the criteria to be traded for ERC-1155 or ERC-721 and in some cases where need be they could present themselves as BEP-2 ERC-20 tokens for efficiency.

Reply:
@okwme that is about implmentation, as it says in the mint token [operation](https://github.com/binance-chain/BEPs/pull/7/files/bafb62d95b4d455d6bc41f755316524b5b91dc7a#diff-03fa5f1a99f35738a833285aee92fa25R189): 

```
Issue token

 This can be performed only when the following conditions are satisfied:
 1. Executor is the collection owner.
 2. The amount of the minted tokens on the collections has not exceeded the total supply of the collection.
```